plugins/modules/cloud/misc/proxmox_interface.py

```diff
@@ -0,0 +1,332 @@
+    type: str
+    required: true
+  state:
+    description:
+      - Indicates desired state of the NIC.
+    type: str
+    choices:
+      - present
+      - absent
+      - gathered
+    default: present
+extends_documentation_fragment:
+  - community.general.proxmox.documentation
+'''
+
+EXAMPLES = '''
+- name: Create new VM bridge on node01
+  community.general.proxmox_interface:
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    node: node01
+    type: bridge
+    name: vmrb56
+    address: 192.168.5.4
+    netmask: 255.255.255.0
+    comment: 'This is a new VM bridge with the host inside'
+    state: present
+
+- name: Delete interface vmbr123 from node01
+  community.general.proxmox_interface:
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    name: vmbr123
+    node: node01
+    state: absent
+
+- name: Gather info about interface eth1 from node01
+  community.general.proxmox_interface
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    name: eth1
+    node: node01
+    state: gathered
+'''
+
+RETURN = '''
+
+config:
+  description: The configuration as structured data before module execution
+  returned: always
+  type: list
+  sample:
+   - [{'exists': 1, 'type': 'eth', 'families': ['inet', 'inet6'], 'method6': 'static', 'netmask6': '64', 'address': '10.0.0.12', 'priority': 4, 'active': 1,
+   - 'cidr6': 'abcd:abcd:abcd::2/64', 'gateway6': 'fe80::1', 'autostart': 1, 'method': 'static', 'address6': 'abcd:abcd:ab:abcd::2', 'netmask': '26',
+   - 'iface': 'ens32', 'gateway': '10.0.0.1', 'cidr': '10.0.0.10/24'}]
+
+config_after:
+  description: The configuration as structured data after module completion
+  returned: when changed
+  type: list
+  sample: same as in config
+'''
+
+
+def reload_interfaces(proxmox: ProxmoxAnsible):
+    if proxmox.module.params['apply']:
+        try:
+            proxmox.proxmox_api.nodes(
+                proxmox.module.params['node']).network.put()
+        except Exception as e:
+            proxmox.module.fail_json(msg="Failed to reload and apply interfaces on node {0}"
+                                     + " with exception: {1}".format(proxmox.module.params['node'], str(e)))
+
+
+def main():
+    interface_common_args = proxmox_interface_argument_spec()
+    network_args = dict(
+        apply=dict(type=bool, default=True),
+        node=dict(type='str', requied=True)
+    )
+    module_args = proxmox_auth_argument_spec()
+    module_args.update(network_args)
+    module_args.updated(interface_common_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_together=[('api_token_id', 'api_token_secret'),
+                           ('api_user', 'api_password'),
+                           ('state', 'present', ('config'))],
+        required_one_of=[('api_password', 'api_token_id')],
+        required_if=[('state', 'present', ('config',))],
+        supports_check_mode=True,
```

Comment:
The module does not seem to actually implement check mode. (You should implement it, instead of setting this to `False`.)